pkg/kubernetes/nulecule.js

```diff
@@ -85,6 +87,98 @@ define([
+                    var message = _("Unable to fetch specversion from image.");
+                    var err = new Error(message);
+                    dfd.reject(err);
+                })
+                .done(function(data) {
+                    var datal = data.split("\n");
+                    for (var i = datal.length - 1; i >= 0; i--) {
+                        if(datal[i].indexOf("io.projectatomic.nulecule.atomicappversion:") > -1) {
+                            msgl = datal[i].split('io.projectatomic.nulecule.atomicappversion:');
+                            if(msgl.length > 1)
+                                versions.atomicappversion = msgl[1];
+                        } else if (datal[i].indexOf("io.projectatomic.nulecule.specversion:") > -1) {
+                            msgl = datal[i].split('io.projectatomic.nulecule.specversion:');
+                            if(msgl.length > 1)
+                                versions.specversion = msgl[1];
+                        } else if(datal[i].indexOf("io.projectatomic.nulecule.providers:") > -1) {
+                            msgl = datal[i].split('io.projectatomic.nulecule.providers:');
+                            if(msgl.length > 1)
+                                versions.providers = msgl[1];
+                        }
+                    }
+                    debug(versions);
+                    if (Object.keys(versions).length == 3) {
+                        dfd.resolve(versions);
+                    } else {
+                        var message;
+                        if (!versions.providers)
+                            message = _("This image does not contain io.projectatomic.nulecule.providers .");
+                        else if (!versions.specversion)
+                            message = _("This image does not contain io.projectatomic.nulecule.specversion.");
+                        else if (!versions.atomicappversion)
+                            message = _("This image does not contain io.projectatomic.nulecule.atomicappversion.");
+                        debug("missing version data in image");
+                        var err = new Error(message);
+                        dfd.reject(err);
+                    }
+                })
+                .always(function(){
+                    process.close();
+                });
+
+            var promise = dfd.promise();
+            promise.cancel = function cancel() {
+                process.close();
+            };
+
+            return promise;
+        }
+
+        function check_atomic_version() {
+            var dfd = $.Deferred();
+            var msgl;
+            var versions = {};
+            debug("checking atomic version");
+            var process = cockpit.spawn(["/usr/bin/atomic", "--version"],{ err: "out", superuser: true , pty: true })
+                .fail(function(ex) {
+                    console.warn(ex);
+                    var message = _("atomic is not installed.");
```

Comment:
This is way to abstract. Lets say: "The 'atomic' command is not installed on the system."